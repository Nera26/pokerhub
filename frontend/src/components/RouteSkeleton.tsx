import RouteLoading from '@/components/RouteLoading';
import SkeletonGrid from '@/app/components/common/SkeletonGrid';
import type { PropsWithChildren } from 'react';

interface RouteSkeletonProps {
  className?: string;
  wrapperClassName?: string;
  rows?: number;
  cardHeight?: string;
}

export default function RouteSkeleton({
  children,
  className,
  wrapperClassName = 'animate-pulse space-y-6',
  rows,
  cardHeight,
}: PropsWithChildren<RouteSkeletonProps>) {
  return (
    <RouteLoading className={className}>
      <div className={wrapperClassName}>
        {children}
        <SkeletonGrid rows={rows} cardHeight={cardHeight} />
      </div>
    </RouteLoading>
  );
}
