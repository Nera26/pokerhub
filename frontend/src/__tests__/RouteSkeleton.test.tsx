import { render } from '@testing-library/react';
import RouteSkeleton from '@/components/RouteSkeleton';

describe('RouteSkeleton', () => {
  it('renders wrapper, children, and grid', () => {
    const { container } = render(
      <RouteSkeleton rows={1} cardHeight="h-10">
        <div className="placeholder" />
      </RouteSkeleton>,
    );
    expect(container).toMatchSnapshot();
  });

  it('renders wrapper and grid without children', () => {
    const { container } = render(<RouteSkeleton rows={1} cardHeight="h-10" />);
    expect(container).toMatchSnapshot();
  });
});
