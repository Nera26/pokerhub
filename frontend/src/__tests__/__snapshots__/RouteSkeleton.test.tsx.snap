// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteSkeleton renders wrapper and grid without children 1`] = `
<div>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
  <main
    aria-busy="true"
    class="container mx-auto px-4 sm:px-6 lg:px-8 pt-6 md:pt-8 pb-[calc(env(safe-area-inset-bottom)+72px)]"
  >
    <div
      class="animate-pulse space-y-6"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 sm:gap-6"
      >
        <div
          aria-label="Loading table"
          class="rounded-2xl bg-card-bg animate-pulse h-10"
        />
      </div>
    </div>
  </main>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
</div>
`;

exports[`RouteSkeleton renders wrapper, children, and grid 1`] = `
<div>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
  <main
    aria-busy="true"
    class="container mx-auto px-4 sm:px-6 lg:px-8 pt-6 md:pt-8 pb-[calc(env(safe-area-inset-bottom)+72px)]"
  >
    <div
      class="animate-pulse space-y-6"
    >
      <div
        class="placeholder"
      />
      <div
        class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-4 sm:gap-6"
      >
        <div
          aria-label="Loading table"
          class="rounded-2xl bg-card-bg animate-pulse h-10"
        />
      </div>
    </div>
  </main>
  <div
    class="h-20 bg-card-bg animate-pulse"
  />
</div>
`;
